tests/functional/test_error_handling.py

```diff
@@ -1,25 +1,15 @@
 import os
 import re
+from concurrent.futures.process import BrokenProcessPool
 
 import pytest  # type: ignore
-
 from grimp import build_graph, exceptions
 
 
-def test_syntax_error_includes_module():
-    dirname = os.path.dirname(__file__)
-    filename = os.path.abspath(
-        os.path.join(dirname, "..", "assets", "syntaxerrorpackage", "foo", "one.py")
-    )
-
-    with pytest.raises(exceptions.SourceSyntaxError) as excinfo:
+def test_syntax_error_terminates_executor_pool():
+    with pytest.raises(BrokenProcessPool):
```

Comment:
I guess this is one disadvantage of concurrency in python. using the .map() method in the executor results in the inability to capture errors raised from individual tasks

Reply:
It would be a shame to lose that exception - returning any syntax errors (they could even be exception objects) sounds like it is at least worth a try.